frontend/src/views/admin/assignments/action-buttons.tsx

```diff
@@ -0,0 +1,63 @@
+import React from "react";
+import {
+    FaEnvelope,
+    FaBan,
+    FaCheck,
+    FaUserTimes,
+    FaUserClock,
+    FaUserPlus,
+} from "react-icons/fa";
+import { ActionButton } from "../../../components/action-buttons";
+
+export function ActionButtons(props: {
+    actionPermitted: { [Key: string]: boolean };
```

Comment:
I was suggesting you have one object per action with a button/dialog grouped, not one object for buttons and one for dialogs.